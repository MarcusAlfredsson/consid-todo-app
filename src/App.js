import React, { Component } from 'react';
import logo from './logo.svg';
import * as style from './App.module.css';

class App extends Component {
  render() {
    return (
      <div className={style.App}>
        <div className={style['App-header__title']}>
          Todo app
        </div>
        <div className={style['App-header__symbol']}>
          <img src={logo} className={style['App-logo']} alt="logo" />
        </div>
        <div className={style.content}>
        test content
        </div>
      </div>
    );
  }
}

export default App;
