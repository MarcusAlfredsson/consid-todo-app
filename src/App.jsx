import React, { Component } from 'react';
import logo from './logo.svg';
import * as style from './App.module.css';
import * as Api from './Api';

class App extends Component {
  state = {
    todos: [],
    inputText: '',
  };

  onInputChange = (event) => {
    this.setState({ inputText: event.target.value });
  }

  addTodo = () => {
    const { todos } = this.state;
    const newTodos = [ ...todos, this.state.inputText];
    this.setState({
      inputText: '',
      todos: newTodos
    });
  }

  removeTodo = (todo) => {
    // TODO steg 2: ta bort den todo som kommer in som input i state variablen todos
    // tips, anv√§nd filter.
  }

  render() {
    return (
      <div className={style.App}>
        <div className={style['App-header__title']}>
          Todo app
        </div>
        <div className={style['App-header__symbol']}>
          <img src={logo} className={style['App-logo']} alt="logo" />
        </div>
        <div className={style.content}>
        <div>
          <input type="text" value={this.state.inputText} onChange={this.onInputChange} />
          <button type="button" onClick={this.addTodo}>add me</button>
        </div>
        {this.state.todos.map(todo => (<div key={todo}>{todo}</div>))}
        </div>
      </div>
    );
  }
}

export default App;
