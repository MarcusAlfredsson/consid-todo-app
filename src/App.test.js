import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('app component', () => {
  it('addTodo when called then add new item to todos', () => {
    const app = renderer.create(<App />);
    const componentInstance = app.getInstance();
    componentInstance.setState({ inputText: 'test', todos: [] });
    componentInstance.addTodo();
    expect(componentInstance.state.todos.length).toBe(1);
    expect(componentInstance.state.todos[0]).toBe('test');
  });
  it('snapshot that template adds more todos', () => {
    const app = renderer.create(<App />);
    const componentInstance = app.getInstance();
    componentInstance.setState({ todos: ['test snapshot']});
    expect(app).toMatchSnapshot();
  });
});  

