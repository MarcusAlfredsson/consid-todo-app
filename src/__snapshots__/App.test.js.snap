// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component snapshot that template adds more todos 1`] = `
<div>
  <div>
    Todo app
  </div>
  <div>
    <img
      alt="logo"
      src="logo.svg"
    />
  </div>
  <div>
    <ul>
      <li>
        test snapshot
      </li>
    </ul>
    <div>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        type="button"
      >
        add me
      </button>
    </div>
  </div>
</div>
`;
